api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -109,8 +112,15 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 			setPersonFromEncounter(obs);
 			return saveObsNotDirty(obs, changeMessage);
 		} else {
-			setPersonFromEncounter(obs);
-			return saveExistingObs(obs,changeMessage);
+			Errors errors = new BindException(obs, "obs");
+			new ObsValidator().validate(obs, errors);
```

Comment:
You don't need to call the validator, it's called automatically via AOP advice around the saveObs method.

Reply:
@wluyima Ok I will remove the lines..Thanks